Tensile/ClientExecutable.py

```diff
@@ -38,18 +42,20 @@ def __init__(self, sourceDir, buildDir, **options):
     def generate(self):
 
         args = ['cmake']
-        args += itertools.chain.from_iterable([ ['-D', '{}={}'.format(key, value)] for key,value in self.options.items()])
+        args += ['-G', 'Ninja'] if (os.name == 'nt') else []
+        args += itertools.chain.from_iterable([ ['-D{}={}'.format(key, value)] for key,value in self.options.items()])
         args += [self.sourceDir]
+        args = [cmake_path(arg) for arg in args]
 
         Common.print2(' '.join(args))
         with Common.ClientExecutionLock():
-            subprocess.check_call(args, cwd=Common.ensurePath(self.buildDir))
+            subprocess.check_output(args, stderr=subprocess.STDOUT, cwd=Common.ensurePath(self.buildDir))
 
     def build(self):
-        args = ['make', '-j']
+        args = [('ninja' if (os.name == "nt") else 'make'), f'-j{CPUThreadCount()}']
         Common.print2(' '.join(args))
         with Common.ClientExecutionLock():
-            subprocess.check_call(args, cwd=self.buildDir)
+            subprocess.check_output(args, stderr=subprocess.STDOUT, cwd=self.buildDir)
```

Comment:
I'm not intimately familiar with the differences between check_call and check_output; what's the motivation for this change? This version doesn't seem to capture the output of the child process (at least on Linux), which is a bit annoying and potentially confusing, as it appears that the process has stalled while it's building the client.

Reply:
windows hipcc has a bug which will non-block when check_call. we change it to check_output to block in this line until check_output is completed.